scripts/tasks/fluentui-publish.ts

```diff
@@ -1,33 +1,163 @@
+        '--no-git-reset',
+        '--force-publish',
+        '--registry',
+        argv().registry,
+        bumpType,
+      ];
+      break;
+    case 'canary':
+      if (npmTagForCanary.includes('latest')) {
+        throw new Error(`Canary release cannot be published under latest tag`);
+      }
+      lernaPublishArgs = [
+        'lerna',
+        'publish',
+        'prerelease',
+        "--tag-version-prefix='@fluentui/react-northstar_v'",
+        '--no-push',
+        '--no-git-tag-version',
+        '--no-git-reset',
+        '--force-publish',
+        '--registry',
+        argv().registry,
+        '--dist-tag',
+        npmTagForCanary,
+        '--preid',
+        npmTagForCanary,
+      ];
+      break;
+    default:
+      throw new Error(`lerna publish gets invalid bump type ${bumpType}`);
+  }
+
+  if (skipConfirm) {
+    lernaPublishArgs.push('--yes');
+  }
+
+  logger.info(`Running this command: yarn ${lernaPublishArgs.join(' ')}`);
+
+  const result = spawnSync('yarn', lernaPublishArgs, {
+    cwd: fluentRoot,
+    shell: true,
+    stdio: 'inherit',
+  });
+
+  if (result.status) {
+    throw new Error(result.error?.stack || `lerna publish failed with status ${result.status}`);
+  }
+}
+
+const execCommandSync = (cwd: string, command: string, args: string[]) => {
+  const result = spawnSync(command, args, {
+    cwd,
+    shell: true,
+    stdio: 'pipe',
+    encoding: 'utf-8',
+  });
+  if (result.status) {
+    throw new Error(result.error?.stack || `'${command} ${args.join(' ')}' failed with status ${result.status}`);
+  }
+  return result.stdout;
+};
+
+export function fluentuiPostPublishValidation() {
   return function() {
-    const fluentRoot = path.resolve(findGitRoot(), 'packages', 'fluentui');
-    const lernaPublishArgs = [
-      'lerna',
-      'publish',
-      "--tag-version-prefix='@fluentui/react-northstar_v'", // HEADS UP: also see yarn stats:save in azure-pipelines.perf-test.yml
-      '--no-git-reset',
-      '--force-publish',
-      '--registry',
-      argv().registry,
-      bumpType,
-    ];
-
-    logger.info(`Running this command: yarn ${lernaPublishArgs.join(' ')}`);
-
-    const result = spawnSync('yarn', lernaPublishArgs, {
-      cwd: fluentRoot,
-      shell: true,
-      stdio: 'inherit',
-    });
+    const gitRoot = findGitRoot();
 
-    if (result.status) {
-      throw new Error(result.error?.stack || `lerna publish failed with status ${result.status}`);
+    const branch = execCommandSync(gitRoot, 'git', ['branch', '--show-current']);
+    execCommandSync(gitRoot, 'git', ['fetch', 'origin']);
+    execCommandSync(gitRoot, 'git', ['reset', '--hard', `origin/${branch}`]); // sometimes lerna add gitHead in package.json after release
+
+    // sync fluent version
+    execCommandSync(gitRoot, 'yarn', ['syncpack:fix']);
+    const gitStatus = execCommandSync(gitRoot, 'git', ['status']);
+    if (!gitStatus.includes('nothing to commit, working tree clean')) {
```

Comment:
```suggestion
    const gitStatus = execCommandSync(gitRoot, 'git', ['status', '--porcelain']);
    if (gitStatus.length === 0) {
```

nit, I would rather rely on `--porcelain` for machine readable output:
https://stackoverflow.com/a/6978402/6488546